src/js/components/Notification/Notification.js

```diff
@@ -45,39 +90,92 @@ const Notification = ({ message, onClose, id, status, title, toast }) => {
 
   const { icon: CloseIcon } = theme.notification.close;
-  const { icon: StatusIcon, color } = theme.notification[status];
+  const { icon: StatusIcon, background, color } = theme.notification[status];
   const { color: closeIconColor } = theme.notification.close;
 
+  const { direction } = toast ? theme.notification.toast : theme.notification;
+
+  const TextWrapper = direction === 'row' ? Text : Fragment;
+
+  // notification is built with two child boxes that contain:
+  // 1. icon + text (wrapped in button when clickable)
+  // 2. close button
+  // pad needs to be applied to the child boxes, but we don't want to apply
+  // extra padding between the icon + text and the button.
+  const { pad } = theme.notification.container;
+  let textPad;
+  let closeButtonPad;
+  if (onClose) [textPad, closeButtonPad] = adaptThemeStyle(pad, theme);
+  else textPad = pad;
+
+  let actions;
+  let message = messageProp;
+
+  if (actionsProp)
+    actions = actionsProp.map((action, index) => (
+      <NotificationAnchor
+        key={action.label}
+        // create space between first anchor and text content
+        margin={
+          index === 0 && (message || title) ? { left: 'small' } : undefined
+        }
+        {...action}
+      />
+    ));
+
+  const MessageWrapper = direction !== 'row' ? Paragraph : Text;
+  if (message || actions)
+    message = (
+      <MessageWrapper {...theme.notification.message}>
+        {message}
+        {/* include actions with message so it wraps with message */}
+        {actions}
+      </MessageWrapper>
+    );
+
   let content = (
     <Box
       {...theme.notification.container}
+      {...(global ? { ...theme.notification.global.container } : {})}
       {...(toast ? { ...theme.notification.toast.container } : {})}
+      background={
+        !toast && background
+          ? background
+          : theme.notification.container.background
+      }
+      // let internal box control pad
+      pad={undefined}
       direction="row"
+      gap="small"
     >
-      <Box {...theme.notification.iconContainer}>
-        <StatusIcon color={color} />
-      </Box>
-      <Box
-        {...theme.notification.textContainer}
-        align="start"
-        direction="row"
-        justify="between"
-        flex
-      >
-        <Box>
-          <Text {...theme.notification.title}>{title}</Text>
-          {message && (
-            <Paragraph {...theme.notification.message}>{message}</Paragraph>
-          )}
+      {/* separate from onClose button to allow "onClick" in the future and 
+        avoid nested interactive elements */}
+      <Box direction="row" pad={textPad} flex>
+        <Box {...theme.notification.iconContainer}>
+          <StatusIcon color={color} />
+        </Box>
+        <Box {...theme.notification.textContainer}>
+          <TextWrapper>
+            {title && <Text {...theme.notification.title}>{title}</Text>}
+            {message && title && direction === 'row' && <Text> </Text>}
```

Comment:
The DOM would be cleaner if we inserted `<>&nbsp;</>` instead.

I have a feeling you had this before but I didn't like it?

Reply:
Don't think I had before, let me try that.